tests/integration-tests/tests/networking/test_cluster_networking.py

```diff
@@ -144,3 +147,123 @@ def _bastion_factory():
 
     yield _bastion_factory
     cfn_stacks_factory.delete_stack(bastion_stack_name, region)
+
+
+@pytest.mark.usefixtures("enable_vpc_endpoints")
+@pytest.mark.usefixtures("os", "scheduler", "instance")
+def test_cluster_in_no_internet_subnet(
+    region, pcluster_config_reader, clusters_factory, vpc_stack, s3_bucket_factory, test_datadir
+):
+    # This test just creates a cluster in a subnet with no internet and checks that no failures occur
+
+    bucket_name = s3_bucket_factory()
+    bucket = boto3.resource("s3", region_name=region).Bucket(bucket_name)
+    bucket.upload_file(str(test_datadir / "pre_install.sh"), "scripts/pre_install.sh")
+
+    vpc_default_security_group_id = get_default_vpc_security_group(vpc_stack.cfn_outputs["VpcId"], region)
+    cluster_config = pcluster_config_reader(
+        vpc_default_security_group_id=vpc_default_security_group_id, bucket_name=bucket_name
+    )
+    cluster = clusters_factory(cluster_config)
+    assert_that(cluster).is_not_none()
+
+    assert_that(len(get_compute_nodes_instance_ids(cluster.cfn_name, region))).is_equal_to(1)
+
+    # Todo: add test for job submission. This will require a proxy node to connect to the head node of the cluster.
```

Comment:
if not super hard it could be worth asserting CW has log files and clustermgtd log file does not have errors. You might use pcluster log commands to do so.

Also I believe the pcluster dcv-connect would not work unless you are in the same vpc? It's worth to call out this limitation. Same applies to pcluster ssh. Could you test them manually from a host within the same vpc?